docs/project/requirements/architecture.md

```diff
@@ -0,0 +1,147 @@
+# Architecture
+
+Un schéma d'implémentation détaillé sera établi pour décrire l'architecture globale de l'infrastructure, y compris les différents composants, leur interaction et leur déploiement.
+
+
+## Application
+
+
+### Proxy: Traefik
+
+* Application router
+* High-availability by fault tolerance
+* Load-balancing by requests distribution
+  ALB ?
+* Helm chart
+* Version
+
+
+### Application: FastAPI
+
+* Python
+  * Version:
+  * Packaging
+* Project management? Poetry?
+
+
+### Databases: PostgreSQL
+
+* Helm chart
+* Version
+* Master/slave for high availability and read load balancing
+* Master failover
+
+
+## Infrastructure
+
+
+### Network
+
+
+#### Application Load Balancer (ALB)
+
+
+#### Virtual Private Cloud (VPC)
+
+* VPC subnet
+
+
+#### Internet Gateways (IGW)
+
+* Global egress
+
+
+#### Availability Zones (AZ)
+
+* 2 availability zones for high availability
+
+
+#### NAT Gateways (NGW)
+
+* Per AZ egress
+* In public subnet
+
+
+#### Subnets
+
+
+##### Public subnets
+
+* Application access
+* Bastion access
+* NAT gateways ?
+
+
+##### Private subnets
+
+* Database
```

Comment:
Our cluster nodes are running in private subnets.
The database runs within our k8s cluster.

Reply:
OK so, as we don't have web frontend servers, app and db are in the same private subnet which is also the cluster subnet, right?

Reply:
Yes. The kubernetes cluster runs in the private subnets and our apps and dbs run inside the kubernetes cluster.